### ISSUE TITLE
R0102 simplifiable-if-statement error message is confusing

### ISSUE THREAD

**Issue by alexjurkiewicz** [PyCQA/pylint#837]
From the manual:

```
simplifiable-if-statement (R0102):
 	The if statement can be reduced by %s Used when an if statement can be reduced to a boolean conversion of the statement’s test.
```

From the CLI:
```
The if statement can be reduced by assigning bool of test
```

Neither of these made a lot of sense to me. I ended up having to find and read the examples in #698.

Could I suggest the message is made more explicit with an example? Something like:

```
The if statement can be replaced with var = bool(test)
```

**Comment by PCManticore** [comment]
Sure! Can you send a PR, please?

**Comment by pdawyndt** [comment]
This pylint error is wrongly issued on this code snippet:

```python
if char.isupper():
    up = True
else:
    down = True
```

In this case, the suggested refactoring is not valid.

**Comment by KennethNielsen** [comment]
@pdawyndt I seem to have hit upon the case you mention. Do you know if it was ever reported?

**Comment by aknooh** [comment]
Pylint interpreter is correct on issuing a warning for the code @pdawyndt provided. Pylint would like you to rewrite your code as follows:

```
def is_upper(char):                                                                                                                        
    return bool((char.isupper()))  
```
Test Cases:
  ```
                                                                                                
>>> is_upper('a')                                                                                                                              
False
>>> is_upper('A')
True
>>> is_upper('HELLO')
True
>>> is_upper('hello')
False
>>> type(is_upper('hello'))
<class 'bool'>
>>> 

```

**Comment by KennethNielsen** [comment]
@aknooh @pdawyndt saves to different variables, which is a variation of what I also hit
